Flaky GS test

<!--
Thanks for submitting a bug report. Please fill the template below,
otherwise we will not be able to process this bug report.
-->

#### Summary
<!-- Summarize the problem in a few sentences: -->

An GS test related to BasicStation seems to be flaky.

#### Steps to Reproduce
<!-- How can we reproduce the problem? -->

1. Random unfortunately, seems to occur in i386 more often than not.

<!--
Please upload relevant configuration (as .txt).
If you use the command "ttn-lw-stack config", you can redact sensitive config.
-->

#### What do you see now?
<!--
Please paste terminal output, upload logs (as .txt) or upload screenshots.
-->

```
  WARN No cookie hash key configured, generated a random one hash_key=64E0A6EDFCF29D09BD4D7850CF991697390FE26254F0D5AF7B5C2D86506EABE6E6953827780422500663D7EC874C0BA1ABE61DE77C5CDA606F1BF5964AF768FB namespace=web test_name=TestClientTokenAuth
  WARN No cookie block key configured, generated a random one block_key=B67A2B682C13B401592CA33D420B7F8BF80CA0D5CEC44382E8B9F5C19CB45967 namespace=web test_name=TestClientTokenAuth
  WARN No static assets found in any search path namespace=web search_paths=[] test_name=TestClientTokenAuth
  WARN No rights TTL configured                 test_name=TestClientTokenAuth
  WARN No cluster key configured, generated a random one key=5d08f3188d6f09ca242e43fa38c71ccb420f0d01722608921069877961b26978 test_name=TestClientTokenAuth
  INFO Listening for connections                address=:0 namespace=grpc protocol=gRPC test_name=TestClientTokenAuth
  INFO Finished unary call                      duration=129.216µs grpc_method=ListRights grpc_service=ttn.lorawan.v3.GatewayAccess namespace=grpc request_id=01DGHW2EQAKCQAT7GMCV56GN7N test_name=TestClientTokenAuth
  INFO Finished unary call                      duration=74.595µs grpc_method=ListRights grpc_service=ttn.lorawan.v3.GatewayAccess namespace=grpc request_id=01DGHW2EQCJFA3T8R0BRGZZ4M7 test_name=TestClientTokenAuth
  INFO Finished unary call                      duration=49.16µs grpc_method=ListRights grpc_service=ttn.lorawan.v3.GatewayAccess namespace=grpc request_id=01DGHW2EQDGF0AZXF9KXFRDWTC test_name=TestClientTokenAuth
  WARN Failed to connect                        endpoint=traffic error=error:pkg/auth/rights:no_gateway_rights (no rights for gateway `eui-0101010101010101`) namespace=gatewayserver/io/basicstation remote_addr=127.0.0.1:38120 test_name=TestClientTokenAuth
  INFO Finished unary call                      duration=47.539µs grpc_method=ListRights grpc_service=ttn.lorawan.v3.GatewayAccess namespace=grpc request_id=01DGHW2EQFV6KWP9F96H2HH6FB test_name=TestClientTokenAuth
  WARN Failed to connect                        endpoint=traffic error=error:pkg/auth/rights:no_gateway_rights (no rights for gateway `eui-1122334455667788`) namespace=gatewayserver/io/basicstation remote_addr=127.0.0.1:38124 test_name=TestClientTokenAuth
  WARN No cookie hash key configured, generated a random one hash_key=D25D7D7536B966F5B6855BBBB32C5DB6B3F15BCB7F6016D1230CFE72ADEDA7CB96DB4B28C4562D2FB0C8BDEECA37B6C6A347BFAB763E337C67D550628025277D namespace=web test_name=TestDiscover
  WARN No cookie block key configured, generated a random one block_key=783F78A2A730148D2D4DAE990C1A382A2245AF8F9B23E81205D0B4B2BB8857FC namespace=web test_name=TestDiscover
  WARN No static assets found in any search path namespace=web search_paths=[] test_name=TestDiscover
  WARN No rights TTL configured                 test_name=TestDiscover
  WARN No cluster key configured, generated a random one key=b444a96de899cd07cd0dd3c2634e5c212e1f6fe1f7761df245bcd2af62394a05 test_name=TestDiscover
  INFO Listening for connections                address=:0 namespace=grpc protocol=gRPC test_name=TestDiscover
  WARN transport: http2Server.HandleStreams failed to read frame: io: read/write on closed pipe namespace=grpc test_name=TestDiscover
  WARN No cookie hash key configured, generated a random one hash_key=90DDE8B5965456DB9031868F686EEC67DDDECB89BB321207A9F8E205799DBCA336D5AD4FB90BF0625E427F3EA203D4C25D382064298E9DAF2330F3E0256F5388 namespace=web test_name=TestVersion
  WARN No cookie block key configured, generated a random one block_key=4B3081EF5A46F7C8B3D382956444762BBA1C99AB6D5E24C440E875A95169D710 namespace=web test_name=TestVersion
  WARN No static assets found in any search path namespace=web search_paths=[] test_name=TestVersion
  WARN No rights TTL configured                 test_name=TestVersion
  WARN transport: http2Server.HandleStreams failed to read frame: io: read/write on closed pipe namespace=grpc test_name=TestVersion
  WARN No cluster key configured, generated a random one key=8aa8c8ae0bfdf1652673095a6b318ca908df7cf3d5ce2cde38927fbcc617e761 test_name=TestVersion
  INFO Listening for connections                address=:0 namespace=grpc protocol=gRPC test_name=TestVersion
  WARN No cookie hash key configured, generated a random one hash_key=85448D0C2AF32B801C66AE02E90160AE3D0B68E544B929332D82CA1422132711762D443BC38C9DB0384D7EB557CFA476F2D395EDD3DCE72FD460F1FAFE67EB24 namespace=web test_name=TestTraffic
  WARN No cookie block key configured, generated a random one block_key=BC72CF892B54AC990EC723D1AD960E124966D838F6672F1DE5555C4EE3354CE8 namespace=web test_name=TestTraffic
  WARN No static assets found in any search path namespace=web search_paths=[] test_name=TestTraffic
  WARN No rights TTL configured                 test_name=TestTraffic
  WARN No cluster key configured, generated a random one key=0a98883240e56659143f719bebe4c385bc28cfd8bd07f872112476f400c80b06 test_name=TestTraffic
  INFO Listening for connections                address=:0 namespace=grpc protocol=gRPC test_name=TestTraffic
  WARN No cookie hash key configured, generated a random one hash_key=855962A5547D26BB48E1E854733C0FFE60900A1DE0F64DC5596BC7C478F54F35B40524DFBE63AB5D18DF8CD2F6478535B4D60352592DBC88860E78C71B64EB63 namespace=web test_name=TestRTT
  WARN No cookie block key configured, generated a random one block_key=C480E99417DED3AA01C45D367C2910E7FFA126E363166FC164152EB34BEAC66E namespace=web test_name=TestRTT
  WARN No static assets found in any search path namespace=web search_paths=[] test_name=TestRTT
  WARN No rights TTL configured                 test_name=TestRTT
  WARN No cluster key configured, generated a random one key=c2be4bf9607b457e938e267f9f85863ef703a5932fed38b2760a0ae8e6ccfe5a test_name=TestRTT
  INFO Listening for connections                address=:0 namespace=grpc protocol=gRPC test_name=TestRTT
--- FAIL: TestRTT (0.16s)
    --- FAIL: TestRTT/FollowUpTxAck (0.13s)
        doc.go:66: 
            /home/travis/gopath/src/go.thethings.network/lorawan-stack/pkg/gatewayserver/io/basicstationlns/basicstationlns_test.go:1098
            Expected: '1'
            Actual:   '0'
            (Should be equal)
        basicstationlns_test.go:1099: Incorrect Stats entries recorded: 0
```

#### What do you want to see instead?
<!-- Please add some examples or mock-ups if applicable. -->

No failure consistently.

#### Environment
<!--
Your environment: OS/Browser/Gateway/Device/...? Versions? IDs/EUIs?
Paste the output of "ttn-lw-cli version" or "ttn-lw-stack version" if applicable.
-->

77556f5f46dce98768c9520e7ea566a8e22ef679

#### How do you propose to implement this?
<!-- Please think about how this could be fixed. -->

Not sure.

#### Can you do this yourself and submit a Pull Request?
<!-- You can also @mention experts if you need help with this. -->

I can review.

I think I have another one:

```
--- FAIL: TestGatewayServer (5.86s)
    --- FAIL: TestGatewayServer/Traffic/basicstation (0.26s)
        doc.go:66: 
            /home/travis/gopath/src/go.thethings.network/lorawan-stack/pkg/gatewayserver/gatewayserver_test.go:600
            Expected: true
            Actual:   false
        --- FAIL: TestGatewayServer/Traffic/basicstation/NotConnected (0.00s)
            gatewayserver_test.go:601: Expected gateway not to be connected yet, but it is
```

<details>
<summary>Details</summary>

```
  WARN No cookie hash key configured, generated a random one hash_key=B4CFAE7FC9F1E92D1D4E89F117D3A2A2A4B63A607EC53BB0704A481717741E93E678D57A888047E6FECDAE72EB5A438C84562727846DD015A622912697D42E37 namespace=web test_name=TestGatewayServer
  WARN No cookie block key configured, generated a random one block_key=3DF4407FE823E823B5B941FD908C25A5268B46C0EA4CE42C10916148DFF1AC72 namespace=web test_name=TestGatewayServer
  WARN No static assets found in any search path namespace=web search_paths=[] test_name=TestGatewayServer
  WARN No rights TTL configured                 test_name=TestGatewayServer
  WARN No cluster key configured, generated a random one key=45d5d0519cefe73de01accad5a11956967f2c812ede8a5b9bd2febec5073cb26 test_name=TestGatewayServer
  INFO Listening for connections                address=:9187 namespace=grpc protocol=gRPC test_name=TestGatewayServer
  INFO Finished unary call                      duration=3.088048ms grpc_method=GetConcentratorConfig grpc_service=ttn.lorawan.v3.GtwGs namespace=grpc request_id=01DKE1NHDZJ12SE1Q7C9BCHJ16 test_name=TestGatewayServer
  INFO Connected                                gateway_uid=eui-aaee000000000000 grpc_method=LinkGateway grpc_service=ttn.lorawan.v3.GtwGs namespace=gatewayserver/io/grpc protocol=grpc remote_addr=127.0.0.1:41046 request_id=01DKE1NHE3J3235DAV458WB0JX test_name=TestGatewayServer
  WARN Link failed                              error=rpc error: code = DeadlineExceeded desc = context deadline exceeded gateway_uid=eui-aaee000000000000 grpc_method=LinkGateway grpc_service=ttn.lorawan.v3.GtwGs namespace=gatewayserver/io/grpc remote_addr=127.0.0.1:41046 request_id=01DKE1NHE3J3235DAV458WB0JX test_name=TestGatewayServer
  WARN Finished streaming call                  duration=248.922066ms error=error:unknown:unknown (context deadline exceeded) error_correlation_id=108e5f6e64cb4b0ab4370a0ff414291f grpc_code=DeadlineExceeded grpc_method=LinkGateway grpc_service=ttn.lorawan.v3.GtwGs namespace=grpc request_id=01DKE1NHE3J3235DAV458WB0JX test_name=TestGatewayServer
  INFO Disconnected                             gateway_uid=eui-aaee000000000000 grpc_method=LinkGateway grpc_service=ttn.lorawan.v3.GtwGs namespace=gatewayserver/io/grpc remote_addr=127.0.0.1:41046 request_id=01DKE1NHE3J3235DAV458WB0JX test_name=TestGatewayServer
  WARN Finished unary call                      duration=691.323µs error=error:pkg/auth/rights:no_gateway_rights (no rights for gateway `eui-aaee000000000000`) error_correlation_id=6e53186e54664ffd9c043e4985a06f8b error_name=no_gateway_rights error_namespace=pkg/auth/rights grpc_code=PermissionDenied grpc_method=GetConcentratorConfig grpc_service=ttn.lorawan.v3.GtwGs namespace=grpc request_id=01DKE1NHNXYXR0X8KTRTWGCRA8 test_name=TestGatewayServer
  WARN Finished unary call                      duration=587.1µs error=error:pkg/auth/rights:no_gateway_rights (no rights for gateway `invalid-gateway`) error_correlation_id=5eff0e1edb7a490ab2492dd5cddbf5dc error_name=no_gateway_rights error_namespace=pkg/auth/rights grpc_code=PermissionDenied grpc_method=GetConcentratorConfig grpc_service=ttn.lorawan.v3.GtwGs namespace=grpc request_id=01DKE1NHNZHMPM4WCWPP4X5BXS test_name=TestGatewayServer
  INFO Finished unary call                      duration=125.419µs error=error:pkg/ttnpb:identifiers (invalid identifiers) error_cause=error:pkg/errors:validation (invalid `gateway_id`: value does not match regex pattern "^[a-z0-9](?:[-]?[a-z0-9]){2,}$") error_correlation_id=a75317eca1a34226b2caaf21ca0fdfee error_name=identifiers error_namespace=pkg/ttnpb grpc_code=InvalidArgument grpc_method=GetConcentratorConfig grpc_service=ttn.lorawan.v3.GtwGs namespace=grpc request_id=01DKE1NHP1W1434MA66EJQ2QAF test_name=TestGatewayServer
  INFO Finished unary call                      duration=69.349µs error=error:pkg/ttnpb:identifiers (invalid identifiers) error_cause=error:pkg/errors:validation (invalid `gateway_id`: value does not match regex pattern "^[a-z0-9](?:[-]?[a-z0-9]){2,}$") error_correlation_id=eaa4ce2b520d408996e723c264e59e33 error_name=identifiers error_namespace=pkg/ttnpb grpc_code=InvalidArgument grpc_method=GetConcentratorConfig grpc_service=ttn.lorawan.v3.GtwGs namespace=grpc request_id=01DKE1NHP203095V9N62KXV8Z7 test_name=TestGatewayServer
  INFO Finished unary call                      duration=403.275µs error=error:pkg/gatewayserver:not_connected (gateway `eui-aaee000000000000` not connected) error_correlation_id=3a9855e40f00475d97d84d317df5ba01 error_name=not_connected error_namespace=pkg/gatewayserver grpc_code=NotFound grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NHP3TGSSJXA6Q593MX5Q test_name=TestGatewayServer
  INFO Finished unary call                      duration=598.031µs grpc_method=GetConcentratorConfig grpc_service=ttn.lorawan.v3.GtwGs namespace=grpc request_id=01DKE1NHP4J45FVV8PC5N3QSSH test_name=TestGatewayServer
  INFO Connected                                gateway_uid=eui-aaee000000000000 grpc_method=LinkGateway grpc_service=ttn.lorawan.v3.GtwGs namespace=gatewayserver/io/grpc protocol=grpc remote_addr=127.0.0.1:41058 request_id=01DKE1NHP523ZSMH1350X3AQ87 test_name=TestGatewayServer
  INFO Finished unary call                      duration=600.845µs grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NHY827DCGCZ4H80CZGTP test_name=TestGatewayServer
  INFO Finished unary call                      duration=538.236µs grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NHYARF12XDXBZD17Z3P8 test_name=TestGatewayServer
  INFO Finished unary call                      duration=480.434µs grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NHYC2HQ2XDN58TGEF5VK test_name=TestGatewayServer
  INFO Finished unary call                      duration=619.404µs grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NHYKF22C1GW0V978H4ZV test_name=TestGatewayServer
  INFO Finished unary call                      duration=607.492µs grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NJ6P8696DAP8PC6GECXA test_name=TestGatewayServer
  INFO Send downlink message                    gateway_uid=eui-aaee000000000000 grpc_method=LinkGateway grpc_service=ttn.lorawan.v3.GtwGs namespace=gatewayserver/io/grpc remote_addr=127.0.0.1:41058 request_id=01DKE1NHP523ZSMH1350X3AQ87 test_name=TestGatewayServer
  INFO Finished unary call                      duration=503.019µs grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NJ6RW8D90F0DKE68E24Z test_name=TestGatewayServer
  INFO Send downlink message                    gateway_uid=eui-aaee000000000000 grpc_method=LinkGateway grpc_service=ttn.lorawan.v3.GtwGs namespace=gatewayserver/io/grpc remote_addr=127.0.0.1:41058 request_id=01DKE1NHP523ZSMH1350X3AQ87 test_name=TestGatewayServer
  INFO Finished unary call                      duration=435.002µs grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NJ6S8Y5TXDQ4A5BF9NE5 test_name=TestGatewayServer
  INFO Finished streaming call                  duration=532.954299ms error=error:unknown:unknown (context canceled) error_correlation_id=4fb9a9f3db7c4cea82be9861dbf5b17e grpc_code=Canceled grpc_method=LinkGateway grpc_service=ttn.lorawan.v3.GtwGs namespace=grpc request_id=01DKE1NHP523ZSMH1350X3AQ87 test_name=TestGatewayServer
  INFO Disconnected                             gateway_uid=eui-aaee000000000000 grpc_method=LinkGateway grpc_service=ttn.lorawan.v3.GtwGs namespace=gatewayserver/io/grpc remote_addr=127.0.0.1:41058 request_id=01DKE1NHP523ZSMH1350X3AQ87 test_name=TestGatewayServer
  INFO Finished unary call                      duration=666.581µs error=error:pkg/gatewayserver:not_connected (gateway `eui-aaee000000000000` not connected) error_correlation_id=b7dd4f083b724fdeb7d88f62093c12ce error_name=not_connected error_namespace=pkg/gatewayserver grpc_code=NotFound grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NJEV3ENPE08V2YXDP1PS test_name=TestGatewayServer
  INFO Connected                                gateway_uid=eui-aaee000000000000 namespace=gatewayserver/io/mqtt protocol=mqtt remote_addr=127.0.0.1:41964 test_name=TestGatewayServer
  INFO Connected                                gateway_uid=eui-aaee000000000000 namespace=gatewayserver/io/mqtt remote_addr=127.0.0.1:41964 test_name=TestGatewayServer
  INFO Disconnected                             gateway_uid=eui-aaee000000000000 namespace=gatewayserver/io/mqtt remote_addr=127.0.0.1:41964 test_name=TestGatewayServer
  INFO Disconnected                             gateway_uid=eui-aaee000000000000 namespace=gatewayserver/io/mqtt remote_addr=127.0.0.1:41964 test_name=TestGatewayServer
  WARN Failed to setup connection               error=error:pkg/auth/rights:no_gateway_rights (no rights for gateway `eui-aaee000000000000`) namespace=gatewayserver/io/mqtt remote_addr=127.0.0.1:41966 test_name=TestGatewayServer
  WARN Failed to setup connection               error=error:pkg/auth/rights:no_gateway_rights (no rights for gateway `eui-aaee000000000000`) namespace=gatewayserver/io/mqtt remote_addr=127.0.0.1:41968 test_name=TestGatewayServer
  WARN Failed to setup connection               error=error:pkg/auth/rights:no_gateway_rights (no rights for gateway `invalid-gateway`) namespace=gatewayserver/io/mqtt remote_addr=127.0.0.1:41970 test_name=TestGatewayServer
  WARN Failed to setup connection               error=error:pkg/auth/rights:no_gateway_rights (no rights for gateway `invalid-gateway`) namespace=gatewayserver/io/mqtt remote_addr=127.0.0.1:41972 test_name=TestGatewayServer
  INFO Finished unary call                      duration=640.272µs error=error:pkg/gatewayserver:not_connected (gateway `eui-aaee000000000000` not connected) error_correlation_id=7fd978dd1d6349bd8d254c78653331e5 error_name=not_connected error_namespace=pkg/gatewayserver grpc_code=NotFound grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NK6YMK2776DJR9P0JC86 test_name=TestGatewayServer
  INFO Connected                                gateway_uid=eui-aaee000000000000 namespace=gatewayserver/io/mqtt protocol=mqtt remote_addr=127.0.0.1:41976 test_name=TestGatewayServer
  INFO Connected                                gateway_uid=eui-aaee000000000000 namespace=gatewayserver/io/mqtt remote_addr=127.0.0.1:41976 test_name=TestGatewayServer
  INFO Finished unary call                      duration=1.215009ms grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NKF2Q28B5W1R8Y5AE9A2 test_name=TestGatewayServer
  INFO Finished unary call                      duration=663.368µs grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NKF4TW87TAHAJ5CAES2B test_name=TestGatewayServer
  INFO Finished unary call                      duration=471.966µs grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NKF67SRSNWME0C0T0YQ5 test_name=TestGatewayServer
  INFO Finished unary call                      duration=429.182µs grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NKF8RGHPNXWVV8FR1MV8 test_name=TestGatewayServer
  INFO Finished unary call                      duration=578.433µs grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NKQAE4V2276G54D23B7F test_name=TestGatewayServer
  INFO Publish downlink message                 gateway_uid=eui-aaee000000000000 namespace=gatewayserver/io/mqtt remote_addr=127.0.0.1:41976 test_name=TestGatewayServer
  INFO Finished unary call                      duration=661.513µs grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NKQCN1PHG2VF5TQ0TNRZ test_name=TestGatewayServer
  INFO Publish downlink message                 gateway_uid=eui-aaee000000000000 namespace=gatewayserver/io/mqtt remote_addr=127.0.0.1:41976 test_name=TestGatewayServer
  INFO Finished unary call                      duration=434.81µs grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NKQEAFDH5N00PM0QS7BZ test_name=TestGatewayServer
  INFO Disconnected                             gateway_uid=eui-aaee000000000000 namespace=gatewayserver/io/mqtt remote_addr=127.0.0.1:41976 test_name=TestGatewayServer
  INFO Disconnected                             gateway_uid=eui-aaee000000000000 namespace=gatewayserver/io/mqtt remote_addr=127.0.0.1:41976 test_name=TestGatewayServer
  INFO Finished unary call                      duration=786.797µs error=error:pkg/gatewayserver:not_connected (gateway `eui-aaee000000000000` not connected) error_correlation_id=2ab3140c367b46a395c707bda4be3bc6 error_name=not_connected error_namespace=pkg/gatewayserver grpc_code=NotFound grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NKZFXE3YBJ8MZTMF0ZQ6 test_name=TestGatewayServer
  INFO Connected                                gateway_eui=AAEE000000000000 gateway_uid=eui-aaee000000000000 namespace=gatewayserver/io/udp protocol=udp test_name=TestGatewayServer
  INFO Disconnected                             gateway_eui=AAEE000000000000 gateway_uid=eui-aaee000000000000 namespace=gatewayserver/io/udp test_name=TestGatewayServer
  WARN Connect unregistered gateway             gateway_eui=BBFF000000000000 gateway_uid=eui-bbff000000000000 namespace=gatewayserver/io/udp protocol=udp test_name=TestGatewayServer
  INFO Connected                                gateway_eui=BBFF000000000000 gateway_uid=eui-bbff000000000000 namespace=gatewayserver/io/udp protocol=udp test_name=TestGatewayServer
  INFO Disconnected                             gateway_eui=BBFF000000000000 gateway_uid=eui-bbff000000000000 namespace=gatewayserver/io/udp test_name=TestGatewayServer
  INFO Finished unary call                      duration=2.580599ms error=error:pkg/gatewayserver:not_connected (gateway `eui-aaee000000000000` not connected) error_correlation_id=f8d378f09aec422e9e6545fe6051aa9a error_name=not_connected error_namespace=pkg/gatewayserver grpc_code=NotFound grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NN71N895JWG6SSBWKSXD test_name=TestGatewayServer
  INFO Connected                                gateway_eui=AAEE000000000000 gateway_uid=eui-aaee000000000000 namespace=gatewayserver/io/udp protocol=udp test_name=TestGatewayServer
  INFO Finished unary call                      duration=658.114µs grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NNFAW8GB8STC71R4Y1Z8 test_name=TestGatewayServer
  INFO Finished unary call                      duration=509.478µs grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NNFCCPNJJD8ERGXS15B9 test_name=TestGatewayServer
  INFO Finished unary call                      duration=454.558µs grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NNFE4ANR3Z1G800967KM test_name=TestGatewayServer
  INFO Finished unary call                      duration=420.41µs grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NNFFJW9X3XJ5H7RRMDFQ test_name=TestGatewayServer
  INFO Finished unary call                      duration=677.001µs grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NNQJ3WPBRE0KA6Q1BC6F test_name=TestGatewayServer
  INFO Finished unary call                      duration=423.739µs grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NNQM2C669SKW6TATFTEP test_name=TestGatewayServer
  INFO Finished unary call                      duration=455.04µs grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NNQNQ9TABHF919Y6BKY4 test_name=TestGatewayServer
  INFO Disconnected                             gateway_eui=AAEE000000000000 gateway_uid=eui-aaee000000000000 namespace=gatewayserver/io/udp test_name=TestGatewayServer
  INFO Finished unary call                      duration=729.169µs error=error:pkg/gatewayserver:not_connected (gateway `eui-aaee000000000000` not connected) error_correlation_id=3d27fb826be649efa2eed66bc5af150d error_name=not_connected error_namespace=pkg/gatewayserver grpc_code=NotFound grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NPBE20E4MDSP4H53HPP0 test_name=TestGatewayServer
  INFO Connected                                gateway_uid=eui-aaee000000000000 namespace=gatewayserver/io/basicstation protocol=basicstation test_name=TestGatewayServer
  INFO Disconnected                             gateway_uid=eui-aaee000000000000 namespace=gatewayserver/io/basicstation test_name=TestGatewayServer
  INFO Connected                                gateway_uid=eui-aaee000000000000 namespace=gatewayserver/io/basicstation protocol=basicstation test_name=TestGatewayServer
  INFO Finished unary call                      duration=566.39µs grpc_method=GetGatewayConnectionStats grpc_service=ttn.lorawan.v3.Gs namespace=grpc request_id=01DKE1NPVHMJR9P63NF3HNFRMK test_name=TestGatewayServer
  INFO Connected                                gateway_uid=eui-aaee000000000000 namespace=gatewayserver/io/basicstation protocol=basicstation test_name=TestGatewayServer
  INFO Disconnected                             gateway_uid=eui-aaee000000000000 namespace=gatewayserver/io/basicstation test_name=TestGatewayServer
  INFO Disconnected                             gateway_uid=eui-aaee000000000000 namespace=gatewayserver/io/basicstation test_name=TestGatewayServer
  WARN transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:9187->127.0.0.1:41080: use of closed network connection namespace=grpc test_name=TestGatewayServer
--- FAIL: TestGatewayServer (5.86s)
    --- FAIL: TestGatewayServer/Traffic/basicstation (0.26s)
        doc.go:66: 
            /home/travis/gopath/src/go.thethings.network/lorawan-stack/pkg/gatewayserver/gatewayserver_test.go:600
            Expected: true
            Actual:   false
        --- FAIL: TestGatewayServer/Traffic/basicstation/NotConnected (0.00s)
            gatewayserver_test.go:601: Expected gateway not to be connected yet, but it is
FAIL
FAIL	go.thethings.network/lorawan-stack/pkg/gatewayserver	5.901s
```

</details>

--- a/pkg/gatewayserver/gatewayserver_test.go
+++ b/pkg/gatewayserver/gatewayserver_test.go
@@ -571,6 +571,9 @@ func TestGatewayServer(t *testing.T) {
 			}
 		})
 
+		// Wait for gateway disconnection to be processed.
+		time.Sleep(timeout)
+
 		t.Run(fmt.Sprintf("Traffic/%v", ptc.Protocol), func(t *testing.T) {
 			a := assertions.New(t)
 

